ci-operator/step-registry/ipi/conf/vsphere/template/ipi-conf-vsphere-template-commands.sh

```diff
@@ -0,0 +1,58 @@
+#!/bin/bash
+set -o errexit
+set -o nounset
+set -o pipefail
+
+
+echo "$(date -u --rfc-3339=seconds) - sourcing context from vsphere_context.sh..."
+# shellcheck source=/dev/null
+declare vsphere_datacenter
+declare vsphere_datastore
+declare vsphere_url
+declare vsphere_cluster
+source "${SHARED_DIR}/vsphere_context.sh"
+# shellcheck source=/dev/null
+source "${SHARED_DIR}/govc.sh"
+if [ -z "${RHCOS_VM_TEMPLATE}" ]; then
+  installer_bin=$(which openshift-install)
+  ova_url=$("${installer_bin}" coreos print-stream-json | jq -r '.architectures.x86_64.artifacts.vmware.formats.ova.disk.location')
+  echo "${ova_url}" > "${SHARED_DIR}"/ova_url.txt
+  vm_template="${ova_url##*/}"
+
+# select a hardware version for testing
+  vsphere_version=$(govc about -json | jq -r .About.Version | awk -F'.' '{print $1}')
+  hw_versions=(15 17 18 19)
+  if [[ ${vsphere_version} -eq 8 ]]; then
+      hw_versions=(20)
+  fi
+  hw_available_versions=${#hw_versions[@]}
+  selected_hw_version_index=$((RANDOM % ${hw_available_versions}))
+  target_hw_version=${hw_versions[$selected_hw_version_index]}
+  echo "$(date -u --rfc-3339=seconds) - Selected hardware version ${target_hw_version}"
+  vm_template=${vm_template}-hw${target_hw_version}
+  
+  ENABLE_TEMPLATE="${SHARED_DIR}"/enable_template_content.txt
+  VERSION=$(echo "${JOB_NAME}" | grep -o -E '4\.[0-9]+')
```

Comment:
Above two lines need to be moved out of `if` block? otherwise, it has issues when defining env `RHCOS_VM_TEMPLATE` explicitly?

Reply:
agree.

Reply:
btw, @jinyunma for vsphere jobs, always using `JOB_NAME` to determinate the cluster version?

Reply:
If the job is added into upgrade ci, I think it would have some problem.
E.g:
```
JOB_NAME="periodic-ci-openshift-openshift-tests-private-release-4.12-amd64-nightly-4.12-upgrade-from-stable-4.12-aws-ipi-network-mtu-localzone-sdn-p2-f28"
VERSION=$(echo "${JOB_NAME}" | grep -o -E '4\.[0-9]+')
echo $VERSION
4.12 4.12 4.12
```

Reply:
good point, only searched relevant code in step [ipi-conf-vsphere](https://github.com/openshift/release/blob/master/ci-operator/step-registry/ipi/conf/vsphere/ipi-conf-vsphere-commands.sh#L34), and this part is indeed executed failed in [upgrade job](https://gcsweb-qe-private-deck-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/qe-private-deck/logs/periodic-ci-openshift-openshift-tests-private-release-4.13-amd64-nightly-4.13-upgrade-from-stable-4.13-vsphere-ipi-proxy-workers-rhel8-p2-f28/1689392905174650880/artifacts/vsphere-ipi-proxy-workers-rhel8-p2-f28/ipi-conf-vsphere/build-log.txt)
```
2023-08-09 21:52:03+00:00 - sourcing context from vsphere_context.sh...
/bin/bash: line 39: [: too many arguments
2023-08-09 21:52:03+00:00 - unable to determine y stream, assuming this is master
2023-08-09 21:52:03+00:00 - 4.x installation is later than 4.9, will install with resource pool
2023-08-09 21:52:03+00:00 - pull-through cache force disabled
```
It is not safe to use this way to get version, might change to another way, e.g from release image.

Reply:
Yeah, I think so.